### PR TITLE
Don't count delivery failure as error

### PR DESCRIPTION
**Pull request by tylerkaraszewski**
### Details
When we re-queue a GetJob(s) command after failing to return it to the client, don't count that as an error.

### Fixed Issues
First half of https://github.com/Expensify/Expensify/issues/211978

### Tests
Test added.

### CODE REVIEW

File: BedrockServer.cpp [hunk 1]
```diff
@@ -1576,8 +1576,10 @@ void BedrockServer::_reply(unique_ptr<BedrockCommand>& command) {
     // Finalize timing info even for commands we won't respond to (this makes this data available in logs).
     command->finalizeTimingInfo();
 
-    // Don't reply to commands with pseudo-clients (i.e., commands that we generated by other commands).
+    // Don't reply to commands with pseudo-clients (i.e., commands that we generated by other commands, or using
+    // `Connection: forget`.
     if (command->initiatingClientID < 0) {
+        command->handleFailedReply();
```
Comment (tylerkaraszewski) [inline BedrockServer.cpp:1582]: We set initiatingClientID to `-1` for `Connection: forget` here: https://github.com/Expensify/Bedrock/blob/434fce36e68179f5476f3c21c772d1dd510e2613/BedrockServer.cpp#L2228

So we should call `handleFailedReply` here for that case.

File: plugins/Jobs.cpp [hunk 2]
```diff
@@ -848,7 +848,7 @@ void BedrockJobsCommand::process(SQLite& db) {
                                          "lastRun = " + SQ(currentTime) + ", " +
                                          "nextRun = " + nextRunDateTime + ", " +
                                          "data = JSON_SET(data, '$.retryAfterCount', COALESCE(JSON_EXTRACT(data, '$.retryAfterCount'), 0) + 1" + // Set this so we don't retry infinitely (see above)
-                                         (isRepeatBasedOnScheduledTime ? ", '$.originalNextRun', " + SQ(job["nextRun"]) + ") ": ")") + // Set this so we don't lose track of the original nextRun (which we are overriding here)
+                                         (isRepeatBasedOnScheduledTime ? ", '$.originalNextRun', " + SQ(job["nextRun"]) + ") ": ") ") + // Set this so we don't lose track of the original nextRun (which we are overriding here)
```
Comment (tylerkaraszewski) [inline plugins/Jobs.cpp:851]: Without the space, the closing paren touches the `WHERE`

File: BedrockServer.cpp [hunk 3]
```diff
@@ -1576,8 +1576,10 @@ void BedrockServer::_reply(unique_ptr<BedrockCommand>& command) {
     // Finalize timing info even for commands we won't respond to (this makes this data available in logs).
     command->finalizeTimingInfo();
 
-    // Don't reply to commands with pseudo-clients (i.e., commands that we generated by other commands).
+    // Don't reply to commands with pseudo-clients (i.e., commands that we generated by other commands, or using
+    // `Connection: forget`.
```
Comment (flodnv) [inline BedrockServer.cpp:1580]:
```suggestion
    // `Connection: forget`)
```

(but also I'm not a fan of breaking a line for 20 chars which can fit on modern screens)

**Review by flodnv — APPROVED** [review]
Thanks!

File: BedrockServer.cpp (hunk 3 above)
Comment (tylerkaraszewski) [reply BedrockServer.cpp:1580]: My editor auto line breaks at 120 chars. Are we not doing that anymore? I can turn it off, it's done that for like 15 years.

Comment (flodnv) [reply BedrockServer.cpp:1580]: Heh, we had a discussion about this somewhere some day. https://expensify.slack.com/archives/C03TQ48KC/p1627427210244000
https://expensify.slack.com/archives/C03TQ48KC/p1627495279311100

Soooo I guess you can make it 190? 😄

**Pull request merged**